avocado/query/utils.py

```diff
@@ -0,0 +1,138 @@
+import logging
+from django.db import connections, DEFAULT_DB_ALIAS
+from django.core.cache import cache
+
+logger = logging.getLogger(__name__)
+
+
+TEMP_DB_ALIAS_PREFIX = '_db:{0}'
+
+
+def named_connection(name, db=DEFAULT_DB_ALIAS):
+    """Opens a new and isolated connection for the named queryset.
+
+    The name is used to create a cache key for storing the backend connection
+    ID which enables it to be canceled later.
+
+    The implementation of this is a tad conflated due to how Django manages
+    connections.
+    """
+    # Define a new database alias
+    temp_db = TEMP_DB_ALIAS_PREFIX.format(name)
+
+    if temp_db in connections.databases:
+        conn = connections[temp_db]
+
+        if conn.is_dirty():
+            raise KeyError('Connection {0} is in a dirty state.'.format(name))
+    else:
+        # Get the settings of the real database
+        connections.ensure_defaults(db)
+        settings = connections.databases[db]
+
+        # Add new database entry into connections handler so when the query
+        # executes the new connection will be accessible.
+        connections.databases[temp_db] = settings
+
+    # Get the connection to the temporary database.
+    conn = connections[temp_db]
```

Comment:
Nit. Move inside `else` block to remove redundant assignment of `conn` in case `temp_db in connections.databases` is True?